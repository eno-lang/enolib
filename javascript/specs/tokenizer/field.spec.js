const { inspectTokenization } = require('./util.js');

const input = `
key: value

key:    value

key    : value

    key    :    value

key: value
`.trim();

describe('Field tokenization', () => {
  it('performs as specified', () => {
    expect(inspectTokenization(input)).toMatchSnapshot();
  });
});
