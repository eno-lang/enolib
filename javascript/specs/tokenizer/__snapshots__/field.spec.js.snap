// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field tokenization performs as specified 1`] = `
Object {
  "_document": Object {
    "depth": 0,
    "elements": Array [
      Object {
        "continuations": Array [],
        "key": "key",
        "line": 0,
        "parent": [Circular],
        "ranges": Object {
          "fieldOperator": Array [
            3,
            4,
          ],
          "key": Array [
            0,
            3,
          ],
          "line": Array [
            0,
            10,
          ],
          "value": Array [
            5,
            10,
          ],
        },
        "type": Symbol(Field),
        "value": "value",
      },
      Object {
        "continuations": Array [],
        "key": "key",
        "line": 2,
        "parent": [Circular],
        "ranges": Object {
          "fieldOperator": Array [
            15,
            16,
          ],
          "key": Array [
            12,
            15,
          ],
          "line": Array [
            12,
            25,
          ],
          "value": Array [
            20,
            25,
          ],
        },
        "type": Symbol(Field),
        "value": "value",
      },
      Object {
        "continuations": Array [],
        "key": "key",
        "line": 4,
        "parent": [Circular],
        "ranges": Object {
          "fieldOperator": Array [
            34,
            35,
          ],
          "key": Array [
            27,
            30,
          ],
          "line": Array [
            27,
            41,
          ],
          "value": Array [
            36,
            41,
          ],
        },
        "type": Symbol(Field),
        "value": "value",
      },
      Object {
        "continuations": Array [],
        "key": "key",
        "line": 6,
        "parent": [Circular],
        "ranges": Object {
          "fieldOperator": Array [
            54,
            55,
          ],
          "key": Array [
            47,
            50,
          ],
          "line": Array [
            43,
            64,
          ],
          "value": Array [
            59,
            64,
          ],
        },
        "type": Symbol(Field),
        "value": "value",
      },
      Object {
        "continuations": Array [],
        "key": "key",
        "line": 8,
        "parent": [Circular],
        "ranges": Object {
          "fieldOperator": Array [
            69,
            70,
          ],
          "key": Array [
            66,
            69,
          ],
          "line": Array [
            66,
            76,
          ],
          "value": Array [
            71,
            76,
          ],
        },
        "type": Symbol(Field),
        "value": "value",
      },
    ],
    "type": Symbol(Document),
  },
  "_lineCount": 9,
  "_meta": Array [],
}
`;
